### DISCUSSION TITLE
Retrieve GitHub OAuth token

### DISCUSSION THREAD

**Discussion by aaryamvn** [nextauthjs/next-auth#1838] [Help] [answered]
I have an application that revolves around the GitHub API. Therefore it must be obvious I need an OAuth token to access it, but I don't know how to retrieve it, as it is not on the `session.user` object. I am trying to configure a custom provider and providing Github Information, but don't know *exactly* what to put. I would really appreciate it if someone could tell me what object to pass in. Essentially replicating what is on [this](https://next-auth.js.org/configuration/providers#using-a-custom-provider) site, but replacing the Google values with GitHub equivalents, and also adding a way to access the OAuth token on the client. Thanks in advance!

**Comment by balazsorban44** [comment, marked as answer]
Check the jwt and session callbacks

https://next-auth.js.org/configuration/callbacks

**Reply by aaryamvn** [reply]
Hm okay, is the `getSession()` method any valuable? I was reading the docs and noticed it did indeed contain the oauth token, so im assuming i can pass this into the github api to recieve stuff?

**Reply by balazsorban44** [reply]
We support GitHub as a built-in provider: https://next-auth.js.org/providers/github

If you need the access token server-side, use `getSession`. if you need it client-side, use `useSession`

**Reply by aaryamvn** [reply]
I tried doing the same, however, it doesn't return it to me.

**Reply by balazsorban44** [reply]
if you have issues, please provide a reproduction. otherwise we cannot help you

**Reply by aaryamvn** [reply]
it's fine, i think ive found a workaround. Thank you!

**Reply by balazsorban44** [reply]
feel free to share it, in case someone else stumbled upon this discussion!, 😊
